src/app/plugins/marketplace.plugin.ts

```diff
@@ -14,7 +14,7 @@ export class MarketPlacePlugin extends Plugin {
   private _LISTING_PREFIX = '!marketplace add';
   private _ALIAS_PREFIX = '!market add';
   private _TARGET_REACTION = '💰';
-  private _lastListingPost: Maybe<IMessage> = undefined;
+  private _lastListingPost: Maybe<IMessage[]> = undefined;
```

Comment:
nit: since you use an array now you could just use empty array as your 'undefined' value